src/DDTrace/Integrations/Laravel/LaravelIntegration.php

```diff
@@ -186,16 +184,34 @@ function () use ($rootSpan, $integration) {
                     Tag::RESOURCE_NAME,
                     !empty($_SERVER['argv'][1]) ? 'artisan ' . $_SERVER['argv'][1] : 'artisan'
                 );
-                return false;
             }
         );
 
-        \DDTrace\trace_method(
+        \DDTrace\hook_method(
             'Symfony\Component\Console\Application',
             'renderException',
-            function (SpanData $span, $args) use ($rootSpan) {
+            function ($This, $scope, $args) use ($rootSpan) {
                 $rootSpan->setError($args[0]);
-                return false;
+            }
+        );
+
+        \DDTrace\hook_method(
+            'Illuminate\Foundation\Http\Kernel',
+            'renderException',
+            function ($This, $scope, $args) use ($rootSpan) {
+                if (!$rootSpan->hasError()) {
+                    $rootSpan->setError($args[1]);
+                }
```

Comment:
I have no particularly reason to wrap this in `hasError`, but I was thinking that if the error was already set then we'd want to avoid clobbering it. Feel free to remove, I don't feel strongly about this -- just wanted it here for discussion.

Reply:
I would've wrapped it as well. Usually the first failure is the most important one, everything behind may just be part of a cascading failure. Should definitely be left in here.